src/spicy/spicyz/glue-compiler.cc

```diff
@@ -134,23 +131,24 @@ static std::string extract_string(const std::string& chunk, size_t* i) {
     return str;
 }
 
-static hilti::Type extract_type(const std::string& chunk, size_t* i) {
+static hilti::UnqualifiedTypePtr extract_type(Builder* builder, const std::string& chunk, size_t* i) {
     eat_spaces(chunk, i);
 
     // We currently only parse Spicy types that can appear in parameters of
     // built-in hooks--which are not many.
     auto token = extract_id(chunk, i);
 
     if ( token == hilti::ID("string") )
-        return hilti::type::String();
+        return builder->typeString();
 
     if ( token == hilti::ID("uint64") )
-        return hilti::type::UnsignedInteger(64);
+        return builder->typeUnsignedInteger(64);
 
     throw ParseError("mismatching type");
 }
 
-static hilti::type::function::Parameter extract_parameter(const std::string& chunk, size_t* i) {
+static std::shared_ptr<hilti::type::function::Parameter> extract_parameter(Builder* builder, const std::string& chunk,
```

Comment:
Can we use the type alias in `declaration` here? It resolves to exactly the same type.

```suggestion
static hilti::type::function::ParameterPtr extract_parameter(Builder* builder, const std::string& chunk,
```